internal/providers/azure/azure.go

```diff
@@ -66,29 +63,43 @@ func FetchConfig(f *resource.Fetcher) (types.Config, report.Report, error) {
 // FetchFromOvfDevice has the return signature of platform.NewFetcher. It is
 // wrapped by this and AzureStack packages.
 func FetchFromOvfDevice(f *resource.Fetcher, ovfFsTypes []string) (types.Config, report.Report, error) {
-	devicePath := filepath.Join(distro.DiskByIDDir(), configDeviceID)
-
+	var rawConfig []byte
 	logger := f.Logger
-	logger.Debug("waiting for config DVD...")
-	waitForCdrom(logger, devicePath)
 
-	fsType, err := checkOvfFsType(logger, devicePath, ovfFsTypes)
-	if err != nil {
-		return types.Config{}, report.Report{}, err
+	for _, ovfFsType := range ovfFsTypes {
+		device, err := execUtil.GetBlockDevices(ovfFsType)
+		if err != nil {
+			return types.Config{}, report.Report{}, fmt.Errorf("failed to retrieve block devices with FSTYPE=UDF: %v", err)
+		} else if len(device) > 0 {
+			for i, dev := range device {
+				rawConfig, err = getRawConfig(f, dev)
+				if len(rawConfig) > 0 && isCdromPresent(logger, dev) {
```

Comment:
The `isCdromPresent` call should probably have a comment like `// verify that this is a CD-ROM device`, since it's not obvious why we need to check that a disc is present.

We should be checking that this is a CD-ROM drive before we try mounting and reading it, not after.  Doing it beforehand is cheaper, and also avoids reading data from a potentially untrusted volume.

`len(rawConfig) == 0` is valid; it could mean that the user hasn't specified a config.